### PR TITLE
pdnsutil: Warn if records in a zone are eclipsed

### LINKED ISSUE #3059: pdnsutil output PTR/NS
As requested:

```
pdnssec check-zone 1.168.192.in-addr.arpa
[Error] Following record is auth=0, run pdnssec rectify-zone?: 50.1.168.192.in-addr.arpa IN PTR serv50.example.nl
Checked 260 records of '1.168.192.in-addr.arpa', 1 errors, 0 warnings.
```

Rectifying the `1.168.192.in-addr.arpa` doesn't solve this issue.
That is because the issue is that besides the PTR, a NS record also exists.

```
mysql> SELECT * FROM records WHERE name ="50.1.168.192.in-addr.arpa" AND auth=0;
+--------+-----------+----------------------------+------+---------------------------+------+------+-------------+----------+-----------+------+
| id     | domain_id | name                       | type | content                   | ttl  | prio | change_date | disabled | ordername | auth |
+--------+-----------+----------------------------+------+---------------------------+------+------+-------------+----------+-----------+------+
| 787970 |      8952 | 50.1.168.192.in-addr.arpa | PTR  | serv50.example.nl      | 7200 |    0 |  1143045193 |        0 | 50        |    0 |
| 824355 |      8952 | 50.1.168.192.in-addr.arpa | NS   | ns1.example.com | 7200 |    0 |  1146221788 |        0 | 50        |    0 |
+--------+-----------+----------------------------+------+---------------------------+------+------+-------------+----------+-----------+------+
2 rows in set (0.38 sec)
```

Both records cannot exists together, but the error is only about the PTR.

### PR DESCRIPTION
**Pull request by pieterlexis**
### Short description
Fixes #3059 

This is the output when eclipsing happens:
```
../../pdns/pdns/pdnsutil --config-dir=. check-zone example.com
Oct 30 17:47:31 Reading random entropy from '/dev/urandom'
Oct 30 17:47:31 gsqlite3: connection to '/home/lieter/src/PowerDNS/pdns-conf/3/pdns.db' successful
Oct 30 17:47:31 gsqlite3: connection to '/home/lieter/src/PowerDNS/pdns-conf/3/pdns.db' successful
[Warning] 'eclipsed.example.com|A' in zone 'example.com' is eclipsed by the zone-cut at 'eclipsed.example.com'
[Warning] 'dsfsdfs.fffff.eclipsed.example.com|A' in zone 'example.com' is eclipsed by the zone-cut at 'eclipsed.example.com'
Checked 5 records of 'example.com', 0 errors, 2 warnings.
```

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled and tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by Habbie** [comment]
RFC6672 and 7719 use 'occluded' instead of 'eclipsed'

File: pdns/pdnsutil.cc [hunk 1]
```diff
@@ -508,6 +516,15 @@ int checkZone(DNSSECKeeper &dk, UeberBackend &B, const DNSName& zone, const vect
     }
   }
 
+  for(const auto &nsrecord : nsrecords) {
+    for (const auto &q : recs) {
+      if (q.first.isPartOf(nsrecord)) {
+        cout<<"[Warning] '"<<q.first<<"|"<<q.second.getName()<<"' in zone '"<<zone<<"' is eclipsed by the zone-cut at '"<<nsrecord<<"'"<<endl;
```
Comment (Habbie) [inline pdns/pdnsutil.cc]: RFC6672 and 7719 use 'occluded' instead of 'eclipsed'

Comment (Habbie) [reply pdns/pdnsutil.cc]: DNAMEs also occlude names below them

Comment (Habbie) [reply pdns/pdnsutil.cc]: And maybe say 'delegation' here instead of zone-cut?

File: pdns/pdnsutil.cc [hunk 2]
```diff
@@ -266,7 +266,8 @@ int checkZone(DNSSECKeeper &dk, UeberBackend &B, const DNSName& zone, const vect
 
 
   bool hasNsAtApex = false;
-  set<DNSName> tlsas, cnames, noncnames, glue, checkglue;
+  set<DNSName> tlsas, cnames, noncnames, glue, checkglue, nsrecords;
```
Comment (Habbie) [inline pdns/pdnsutil.cc]: maybe rename `nsrecords` to `delegations` or `children` to make it clear the apex NSset is not included here?

**Comment by pieterlexis** [comment]
Output now looks like this:

```
Oct 30 18:37:08 Reading random entropy from '/dev/urandom'
Oct 30 18:37:08 gsqlite3: connection to '/home/lieter/src/PowerDNS/pdns-conf/3/pdns.db' successful
Oct 30 18:37:08 gsqlite3: connection to '/home/lieter/src/PowerDNS/pdns-conf/3/pdns.db' successful
[Warning] 'eclipsed.example.com|A' in zone 'example.com' is occluded by a delegation at 'eclipsed.example.com'
[Warning] 'dsfsdfs.fffff.eclipsed.example.com|A' in zone 'example.com' is occluded by a delegation at 'eclipsed.example.com'
[Warning] 'record.dname.example.com|A' in zone 'example.com' is occluded by a DNAME at 'dname.example.com'
Checked 7 records of 'example.com', 0 errors, 3 warnings.
```

**Pull request merged**